client/src/components/DataFiles/DataFilesToolbar/DataFilesToolbar.js

```diff
@@ -173,7 +172,8 @@ const DataFilesToolbar = ({ scheme, api }) => {
   const canRename = getFilePermissions('rename', permissionParams);
   const canMove = getFilePermissions('move', permissionParams);
   const canCopy = getFilePermissions('copy', permissionParams);
-  const canTrash = getFilePermissions('trash', permissionParams);
+  const canTrash =
+    getFilePermissions('trash', permissionParams) && !path.startsWith('.Trash');
```

Comment:
__Location for Logic__
The pattern of the surrounding lines of code suggests that all logic for determining this boolean is intended to be managed by `getFilePermissions`. I suggest moving the logic into that method. (See "Reference" for file history.)

<details><summary>Reference</summary>

- [code that introduced `getFilePermissions` did extract logic from here to it](https://github.com/TACC/Core-Portal/commit/5ad8b4d7f8cd39066a9275c4509877013b3a9c89#diff-4bd9eefdef0c64167de1eee994c2ed5efbd75c6a6be1718e34b9220c1b95b341R147)
    - (takes time to load then auto-scroll)
- [the `getFilePermissions` code](https://github.com/TACC/Core-Portal/blob/17ae344/client/src/utils/filePermissions.js)

</details>

Reply:
__Food for Thought__
Should user be prevented from removing files from within all [protected directories](https://github.com/TACC/Core-Portal/blob/17ae344/client/src/utils/filePermissions.js#L19) not just `.Trash`? _I do not have the answer, so [I've asked in #cep-design.](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1631556599030200)_